xapian-applications/omega/handler_gmime.cc

```diff
@@ -109,34 +109,39 @@ parser_content(GMimeObject* me, string& dump)
 	string type = g_mime_content_type_get_media_type(ct);
 	string subtype = g_mime_content_type_get_media_subtype(ct);
 	string charset(g_mime_object_get_headers(me));
-	size_t start = charset.find("charset=");
+	size_t end, start;
+	start = charset.find("charset=");
```

Comment:
Before `start` was initialised at the point of declaration, which nice and clean and generally what you want to do where it's easy to arrange.  If a variable is only declared when you can initialise it there's no point at which is has an undefined value, and that's one less thing to worry about.

But now you don't, and not for any good reason.

In fact `end` would be better declared in the block below where it is used rather than here.

Reply:
Okay, I am going to change it.